src/test/rgw/rgw_multi/zone_rados.py

```diff
@@ -78,7 +83,9 @@ def check_bucket_eq(self, zone_conn, bucket_name):
                 k1_head = b1.get_key(k1.name)
```

Comment:
@cbodley note get_key here doesn't use version_id

Reply:
yeah, k1 is the key that includes the version id. i think the intent here was to check that the 'current version' (according to HEAD on the name itself) matches too. this commit extends that to check whether the current version is a delete marker (k1_head=None), and whether the k2_head matches

Reply:
@cbodley this is the wrong place to check that

Reply:
@cbodley iterating over get_all_versions() here, to check current_version need to iterate over non-versioned bucket listing

Reply:
okay, thanks. so this part should be comparing the version's head objects, and we should compare the olh heads separately.. but a normal bucket listing wouldn't cover the delete marker case. maybe we just do both here? i.e.
```python
# now get the keys through a HEAD operation, verify that the available data is the same
k1_head = b1.get_key(k1.name, k1.version_id)
k2_head = b2.get_key(k2.name, k2.version_id)
check_object_eq(k1_head, k2_head, False)

if k1.version_id:
    # compare the olh to make sure they agree about the current version
    k1_olh = b1.get_key(k1.name)
    k2_olh = b2.get_key(k2.name)
    # if there's a delete marker, HEAD will return None
    if k1_olh or k2_olh:
        check_object_eq(k1_olh, k2_olh, False)
```
the olh check is redundant when there are multiple versions with the same name, but at least that should cover everything